pkg/alerting/templates/funcs.go

```diff
@@ -0,0 +1,194 @@
+package templates
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"strconv"
+	"time"
+
+	amtemplate "github.com/prometheus/alertmanager/template"
+	"github.com/prometheus/common/model"
+)
+
+func floatToTime(v float64) (*time.Time, error) {
+	if math.IsNaN(v) || math.IsInf(v, 0) {
+		return nil, errNaNOrInf
+	}
+	timestamp := v * 1e9
+	if timestamp > math.MaxInt64 || timestamp < math.MinInt64 {
+		return nil, fmt.Errorf("%v cannot be represented as a nanoseconds timestamp since it overflows int64", v)
+	}
+	t := model.TimeFromUnixNano(int64(timestamp)).Time().UTC()
+	return &t, nil
+}
+
+func convertToFloat(i interface{}) (float64, error) {
```

Comment:
Nit: I think @kralicky thinks we should use the `any` alias rather than empty interface.  I do agree that `any` communicates the intent better, but I'm not married to enforcing this.